### PR TITLE
Improve vendor facing doc to collect consent related information. 

### PR DESCRIPTION
**Pull request by zhouyx**
add @jawadst as reviewer as well. Open to suggestions. 

I am seeing this doc as the vendor facing one. 
amp-consent.md will be the publisher facing doc,
and integrating-consent.md will be the CMP facing doc. 

Thank you

### CODE REVIEW

File: extensions/amp-consent/customizing-extension-behaviors-on-consent.md [hunk 1]
```diff
@@ -28,16 +28,41 @@ getConsentPolicy() {
...
+#### If you integrate with AMP as a third party ad vendor
+Access the value within the ad iframe using `window.context.initialConsentState`. Check [this](https://github.com/ampproject/amphtml/blob/master/ads/README.md#amp-consent-integration) for more details.
+
+#### If you integrate with AMP as an analytics vendor
+Get the value using `CONSENT_STATE` macro, or `${consentState}`. A request with the varaible will only be sent out after the state has resolved to one of the above state.
+
+### On Consent String ###
+
+AMP collects raw consent string value from `checkConsentHref` endpoint or from the CMP. The it passes the raw consent string to vendors without modification.
+It is then up to the vendor to intepret the string and customize behavior accordingly. AMP recommends handle the string on the server side.
```
Comment (micajuine-ho) [inline extensions/amp-consent/customizing-extension-behaviors-on-consent.md:55]: nit: s/handle/handling

File: extensions/amp-consent/customizing-extension-behaviors-on-consent.md [hunk 2]
```diff
@@ -28,16 +28,41 @@ getConsentPolicy() {
...
 
-In addition to the consent state, the vendor can also use the
+#### If you integrate with AMP as a first party AMP extension
+Use the `getConsentPolicyState` API. It returns a promise with one of the `CONSENT_POLICY_STATE` value.
+
+#### If you integrate with AMP as a third party ad vendor
+Access the value within the ad iframe using `window.context.initialConsentState`. Check [this](https://github.com/ampproject/amphtml/blob/master/ads/README.md#amp-consent-integration) for more details.
+
+#### If you integrate with AMP as an analytics vendor
+Get the value using `CONSENT_STATE` macro, or `${consentState}`. A request with the varaible will only be sent out after the state has resolved to one of the above state.
```
Comment (tla-sirdata) [inline extensions/amp-consent/customizing-extension-behaviors-on-consent.md:53]: Typo here : `varaible`

File: extensions/amp-consent/customizing-extension-behaviors-on-consent.md [hunk 3]
```diff
@@ -28,16 +28,41 @@ getConsentPolicy() {
...
+
+### On Consent String ###
+
+AMP collects raw consent string value from `checkConsentHref` endpoint or from the CMP. The it passes the raw consent string to vendors without modification.
```
Comment (tla-sirdata) [inline extensions/amp-consent/customizing-extension-behaviors-on-consent.md:57]: Typo here : `The it passes`

File: extensions/amp-consent/customizing-extension-behaviors-on-consent.md (hunk 1 above)
Comment (tla-sirdata) [inline extensions/amp-consent/customizing-extension-behaviors-on-consent.md]: Typo here : `intepret`

**Comment by tla-sirdata** [comment]
Thank you for this vendor facing doc !

**Pull request merged**